frontend/src/components/ProgressBar/ProgressBar.tsx

```diff
@@ -0,0 +1,112 @@
+import React from "react";
+import { useRollbar } from "@rollbar/react";
+import {
+    Button,
+    // LinearProgress
+} from "@mui/material";
+import { useAuth } from "../Authentication/useAuth";
+import styles from "./ProgressBar.module.css";
+
+type Props = {
+    percentage: number;
+};
+
+/**
+ * displays a minimizable progress bar
+ * @param props percentage: a number between 0 and 100, a value not in this range will not display the progress bar
+ * @returns either a progress bar popup with a screen overlay, or nothing
+ * requires percentage prop
+ */
+function ProgressBar(props: Props): React.ReactElement | null {
+    const { isAuthenticated } = useAuth(); // use the useAuth hook to get the current user
+    // If the user is not logged in, don't render the help popup
+    if (!isAuthenticated) {
+        return null;
+    }
+    const rollbar = useRollbar();
+    rollbar.info("ProgressBar");
+    // const [percentage, setPercentage] = useState<number>();
+    const { percentage } = props;
+
+    const openPopup = () => {
+        const popup = document.getElementById("progressBarPopup");
+        const overLay = document.getElementById("progressOverlay");
+        const maxButton = document.getElementById("maximizeButton");
+        console.log(popup);
+
+        if (popup == null || overLay == null || maxButton == null) {
+            rollbar.error("Help popup element does not exist in document");
+        } else {
+            popup.hidden = false;
+            overLay.hidden = false;
+            maxButton.style.display = "none";
+        }
+    };
+
+    const closePopup = () => {
+        const popup = document.getElementById("progressBarPopup");
+        const overLay = document.getElementById("progressOverlay");
+        const maxButton = document.getElementById("maximizeButton");
+        console.log(popup);
+
+        if (popup == null || overLay == null || maxButton == null) {
+            rollbar.error("Help popup element does not exist in document");
+        } else {
+            popup.hidden = true;
+            overLay.hidden = true;
+            maxButton.style.display = "block";
+        }
+    };
+
```

Comment:
To make the progress bar reusable for all components, I think its better to have a prop called isVisible and this should control if the Progress bar should return or return null.

So handling the minimization of the progress bar would be handled outside this component.

Reply:
The open/close popup functions are for handling the minimization requirement of the progress bar. I added the isVisible prop, so it returns null if that is false, or if the percentage < 0 or percentage > 100.